dev/com.ibm.ws.org.apache.jasper.el/src/org/apache/el/util/ReflectionUtil.java

```diff
@@ -178,33 +179,58 @@ public static Method getMethod(EvaluationContext ctx, Object base, Object proper
             int mParamCount = mParamTypes.length;
 
             // Check the number of parameters
-            if (!(paramCount == mParamCount ||
-                    (m.isVarArgs() && paramCount >= mParamCount))) {
+            // Multiple tests to improve readability
+            if (!m.isVarArgs() && paramCount != mParamCount) {
                 // Method has wrong number of parameters
                 continue;
             }
+            if (m.isVarArgs() && paramCount < mParamCount -1) {
+                // Method has wrong number of parameters
+                continue;
+            }
+            if (m.isVarArgs() && paramCount == mParamCount && paramValues != null &&
+                    paramValues.length > paramCount && !paramTypes[mParamCount -1].isArray()) {
+                // Method arguments don't match
+                continue;
+            }
+            if (m.isVarArgs() && paramCount > mParamCount && paramValues != null &&
+                    paramValues.length != paramCount) {
+                // Might match a different varargs method
+                continue;
+            }
+            if (!m.isVarArgs() && paramValues != null && paramCount != paramValues.length) {
+                // Might match a different varargs method
+                continue;
+            }
```

Comment:
Per the comment in the method description, this method was out of sync with the one in Util.java.  I added the relevant portions this method to have it matching again.